_pyansys-dashboard-vulnerabilities-upload/action.yml

```diff
@@ -0,0 +1,191 @@
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+name: >
+  Upload vulnerability reports to PyAnsys Dashboard database.
+
+description: >
+  This action is only intended to be used by repositories that are part of the
+  PyAnsys Dashboard project. It uploads the vulnerability reports generated by
+  the `safety <https://pyup.io/safety/>`_  and `bandit <https://bandit.readthedocs.io/en/latest/>`_
+  tools to the PyAnsys Dashboard database.
+
+inputs:
+
+  # Required inputs
+  pyansys-dashboard-token:
+    description: >
+      Token used to clone the PyAnsys Dashboard repository and upload the
+      vulnerability reports.
+    required: true
+    type: string
+
+  pyansys-dashboard-credentials:
+    description: >
+      Credentials to access the PyAnsys Dashboard database. They
+      are available as a secret inside the ``ansys`` and ``ansys-internal``
+      organizations.
+    required: true
+    type: string
+
+  dry-run:
+    description: >
+      Whether to run or not this action in dry run mode or not.
+      Dry run does not upload the vulnerability reports to the database.
+    required: true
+    type: boolean
+
+  repository:
+    description: >
+      Full name of the repository for which the vulnerability reports are generated.
+    required: true
+    type: string
+
+  safety-report:
+    description: >
+      Path to the safety report file.
+    required: true
+    type: string
+
+  bandit-report:
+    description: >
+      Path to the bandit report file.
+    required: true
+    type: string
+
+  hide-log:
+    description: >
+      Whether to hide the output log of the GitHub action or not.
+      If set to true, the output log will be redirected to /dev/null.
+      If set to false, the output log will be displayed. This would
+      disclose any potential vulnerabilities to anyone.
+    required: true
+    type: boolean
+
+runs:
+  using: "composite"
+  steps:
+    - name: "Check whether the repository owner is 'ansys' or 'ansys-internal'"
+      shell: bash
+      run: |
+        if [[ ${{ github.repository_owner }} != "ansys" && ${{ github.repository_owner }} != "ansys-internal" ]]; then
```

Comment:
This condition should be an "or":

```suggestion
        if [[ ${{ github.repository_owner }} != "ansys" || ${{ github.repository_owner }} != "ansys-internal" ]]; then
```

Reply:
Imagine you have a repo owned by "numpy"

With the original if...

``if [[ numpy != "ansys" && numpy != "ansys-internal" then ...``

Which equals to ``True && True`` --> And consequently the if condition is triggered

With your change, it would be ``True OR True`` which is also valid...

But what happens if it belongs to the "ansys" org?

``False AND True`` --> False --> It does not trigger the IF condition (expected behavior)
``False OR True`` --> True --> It triggers the IF condition (unexpected behavior)

So... @jorgepiloto, I believe your suggestion is wrong

Reply:
Please, verify my thought logic in the previous comment and approve otherwise

Reply:
You are right! I got confused with the negation, my fault. Approving this 👍🏼